src/AfriCC/EPP/Client.php

```diff
@@ -329,49 +329,55 @@ private function recv($length)
+    public function send($buffer)
     {
-        $info = stream_get_meta_data($this->socket);
-        $hard_time_limit = time() + $this->timeout + 2;
-        $length = mb_strlen($buffer, 'ASCII');
-
+        $info = array();
         $pos = 0;
-        while (!$info['timed_out'] && !feof($this->socket)) {
-            // Some servers don't like alot of data, so keep it small per chunk
-            $wlen = $length - $pos;
+        $length = null;
 
-            if ($wlen > 1024) {
-                $wlen = 1024;
-            }
+        if (is_resource($this->socket)) {
+            $info = stream_get_meta_data($this->socket);
+            $hard_time_limit = time() + $this->timeout + 2;
+            $length = mb_strlen($buffer, 'ASCII');
 
-            // try write remaining data from socket
-            $written = @fwrite($this->socket, mb_substr($buffer, $pos, $wlen, 'ASCII'), $wlen);
 
-            // If we read something, bump up the position
-            if ($written) {
-                if ($this->debug) {
-                    $this->log(mb_substr($buffer, $pos, $wlen, 'ASCII'), '1;31');
+            while (!$info['timed_out'] && !feof($this->socket)) {
+                // Some servers don't like alot of data, so keep it small per chunk
+                $wlen = $length - $pos;
+
+                if ($wlen > 1024) {
+                    $wlen = 1024;
                 }
-                $pos += $written;
 
-                // break if all written
-                if ($pos === $length) {
-                    break;
+                // try write remaining data from socket
+                $written = @fwrite($this->socket, mb_substr($buffer, $pos, $wlen, 'ASCII'), $wlen);
+
+                // If we read something, bump up the position
+                if ($written) {
+                    if ($this->debug) {
+                        $this->log(mb_substr($buffer, $pos, $wlen, 'ASCII'), '1;31');
+                    }
+                    $pos += $written;
+
+                    // break if all written
+                    if ($pos === $length) {
+                        break;
+                    }
+                } else {
+                    // sleep 0.25s
+                    usleep(250000);
                 }
-            } else {
-                // sleep 0.25s
-                usleep(250000);
-            }
 
-            // update metadata
-            $info = stream_get_meta_data($this->socket);
-            if (time() >= $hard_time_limit) {
-                throw new Exception('Timeout while writing to EPP Server');
+                // update metadata
+                $info = stream_get_meta_data($this->socket);
+                if (time() >= $hard_time_limit) {
+                    throw new Exception('Timeout while writing to EPP Server');
+                }
             }
         }
 
         // check for timeout
-        if ($info['timed_out']) {
+        if (!array_key_exists('timed_out', $info) || $info['timed_out']) {
```

Comment:
I suppose that @karborator is using code PHP Inspections plugin in Intellij. This one warns about that things. ;)